File: web/src/api/smartStorageCurrentOrder.js
import service from '@/utils/request'

// @Tags SmartStorageCurrentOrder
// @Summary 创建SmartStorageCurrentOrder
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body model.SmartStorageCurrentOrder true "创建SmartStorageCurrentOrder"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"获取成功"}"
// @Router /smartStorageCurrentOrder/createSmartStorageCurrentOrder [post]
export const createSmartStorageCurrentOrder = (data) => {
     return service({
         url: "/smartStorageCurrentOrder/createSmartStorageCurrentOrder",
         method: 'post',
         data
     })
 }


// @Tags SmartStorageCurrentOrder
// @Summary 删除SmartStorageCurrentOrder
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body model.SmartStorageCurrentOrder true "删除SmartStorageCurrentOrder"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"删除成功"}"
// @Router /smartStorageCurrentOrder/deleteSmartStorageCurrentOrder [delete]
 export const deleteSmartStorageCurrentOrder = (data) => {
     return service({
         url: "/smartStorageCurrentOrder/deleteSmartStorageCurrentOrder",
         method: 'delete',
         data
     })
 }

// @Tags SmartStorageCurrentOrder
// @Summary 删除SmartStorageCurrentOrder
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body request.IdsReq true "批量删除SmartStorageCurrentOrder"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"删除成功"}"
// @Router /smartStorageCurrentOrder/deleteSmartStorageCurrentOrder [delete]
 export const deleteSmartStorageCurrentOrderByIds = (data) => {
     return service({
         url: "/smartStorageCurrentOrder/deleteSmartStorageCurrentOrderByIds",
         method: 'delete',
         data
     })
 }

// @Tags SmartStorageCurrentOrder
// @Summary 更新SmartStorageCurrentOrder
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body model.SmartStorageCurrentOrder true "更新SmartStorageCurrentOrder"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"更新成功"}"
// @Router /smartStorageCurrentOrder/updateSmartStorageCurrentOrder [put]
 export const updateSmartStorageCurrentOrder = (data) => {
     return service({
         url: "/smartStorageCurrentOrder/updateSmartStorageCurrentOrder",
         method: 'put',
         data
     })
 }


// @Tags SmartStorageCurrentOrder
// @Summary 用id查询SmartStorageCurrentOrder
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body model.SmartStorageCurrentOrder true "用id查询SmartStorageCurrentOrder"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"查询成功"}"
// @Router /smartStorageCurrentOrder/findSmartStorageCurrentOrder [get]
 export const findSmartStorageCurrentOrder = (params) => {
     return service({
         url: "/smartStorageCurrentOrder/findSmartStorageCurrentOrder",
         method: 'get',
         params
     })
 }


// @Tags SmartStorageCurrentOrder
// @Summary 分页获取SmartStorageCurrentOrder列表
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body request.PageInfo true "分页获取SmartStorageCurrentOrder列表"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"获取成功"}"
// @Router /smartStorageCurrentOrder/getSmartStorageCurrentOrderList [get]
 export const getSmartStorageCurrentOrderList = (params) => {
     return service({
         url: "/smartStorageCurrentOrder/getSmartStorageCurrentOrderList",
         method: 'get',
         params
     })
 }

File: server/router/smartStorageCurrentOrder.go
package router

import (
	v1 "gin-vue-admin/api/v1"

	"github.com/gin-gonic/gin"
)

func InitSmartStorageCurrentOrderRouter(Router *gin.RouterGroup) {
	SmartStorageCurrentOrderRouter := Router.Group("smartStorageCurrentOrder")
	//.Use(middleware.JWTAuth()).Use(middleware.CasbinHandler()).Use(middleware.OperationRecord())
	{
		SmartStorageCurrentOrderRouter.POST("createSmartStorageCurrentOrder", v1.CreateSmartStorageCurrentOrder)             // 新建SmartStorageCurrentOrder
		SmartStorageCurrentOrderRouter.DELETE("deleteSmartStorageCurrentOrder", v1.DeleteSmartStorageCurrentOrder)           // 删除SmartStorageCurrentOrder
		SmartStorageCurrentOrderRouter.DELETE("deleteSmartStorageCurrentOrderByIds", v1.DeleteSmartStorageCurrentOrderByIds) // 批量删除SmartStorageCurrentOrder
		SmartStorageCurrentOrderRouter.PUT("updateSmartStorageCurrentOrder", v1.UpdateSmartStorageCurrentOrder)              // 更新SmartStorageCurrentOrder
		SmartStorageCurrentOrderRouter.GET("findSmartStorageCurrentOrder", v1.FindSmartStorageCurrentOrder)                  // 根据ID获取SmartStorageCurrentOrder
		SmartStorageCurrentOrderRouter.GET("getSmartStorageCurrentOrderList", v1.GetSmartStorageCurrentOrderList)            // 获取SmartStorageCurrentOrder列表
	}
}
